groups/sig-release/groups.yaml

```diff
@@ -150,6 +150,30 @@ groups:
     members:
       - k8s-infra-release-editors@kubernetes.io
 
+  - email-id: k8s-infra-staging-mirror@kubernetes.io
+    name: k8s-infra-staging-mirror
+    description: |-
+      ACL for staging mirror images
+
+      This project is used to stage images mirrored from Docker Hub.
+    settings:
+      ReconcileMembers: "true"
+    members:
+      - k8s-infra-release-editors@kubernetes.io
+      # test-infra-oncall
+      - amerai@google.com
+      - bentheelder@google.com
```

Comment:
baby nit: I'm not in that group anymore (I am in google build admins tho, different rotation)
bigger nit: clarketm is not and is not at google anymore.